### PR TITLE
Fix check in dsymbolsem to compare idents, not toChars()

### PR DESCRIPTION
**Pull request by RazvanN7**
cc @Geod24

### CODE REVIEW

**Comment by Geod24** [comment]
Proper commit message please

**Comment by Geod24** [comment]
Also, see my second message in the original PR.

**Comment by RazvanN7** [comment]
@Geod24 Done, thanks!

File: src/dmd/dsymbolsem.d [hunk 1]
```diff
@@ -3953,7 +3953,7 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
                     {
                         OutBuffer buf1;
 
-                        if (fd.toChars() == funcdecl.toChars())
+                        if (fd.ident.toString == funcdecl.ident.toString())
```
Comment (thewilsonator) [inline src/dmd/dsymbolsem.d:3956]: missing parens.

Comment (RazvanN7) [reply src/dmd/dsymbolsem.d]: Fixed

**Comment by Geod24** [comment]
Wait, don't call `toString` on ident, just compare them! Ident use a pool of strings so that comparison can be fast (it's a pointer comparison instead of `strcmp`). [That was even in Walter's talk](https://youtu.be/bNJhtKPugSQ?t=826) ;)

**Review by Geod24 — APPROVED** [review]

**Pull request merged**